src/pages/physical-count-itemDetails/index.js

```diff
@@ -0,0 +1,616 @@
+              if (parent?.id == 1 && value) return true
+              if (parent?.id == 1 && !value) return true
+              if (parent?.id > 1 && !value) return false
+
+              return value
+            }
+          })
+        })
+      )
+    }),
+    onSubmit: async obj => {
+      const items = obj?.rows?.map((item, index) => ({
+        ...item,
+        seqNo: index + 1,
+        siteId: obj.siteId,
+        stockCountId: obj.stockCountId,
+        controllerId: obj.controllerId
+      }))
+
+      const StockCountItemDetailPack = {
+        siteId: obj.siteId,
+        controllerId: obj.controllerId,
+        stockCountId: obj.stockCountId,
+        items: items.length > 0 && items[0].sku ? items : []
+      }
+
+      await postRequest({
+        extension: SCRepository.StockCountItemDetail.set2,
+        record: JSON.stringify(StockCountItemDetailPack)
+      })
+
+      toast.success(platformLabels.Edited)
+      setEditMode(items.length > 0)
+      checkPhyStatus(obj.controllerId)
+
+      handleClick(items)
+    }
+  })
+
+  async function fetchGridData(controllerId) {
+    const stockCountId = formik.values.stockCountId
+
+    getDTDsku(stockCountId)
+
+    await getRequest({
+      extension: SCRepository.StockCountItemDetail.qry,
+      parameters: `_stockCountId=${stockCountId}&_siteId=${formik.values.siteId}&_controllerId=${controllerId}`
+    }).then(res => {
+      if (res.list) {
+        const modifiedList = res.list?.map((item, index) => ({
+          ...item,
+          id: index + 1,
+          metalPurity: item?.metalPurity || 0,
+          weight: item?.weight || 0,
+          countedQty: item?.countedQty || 0
+        }))
+        modifiedList.length > 0 && formik.setFieldValue('rows', modifiedList)
+      }
+
+      setEditMode(res.list.length > 0)
+      handleClick(res.list)
+    })
+  }
+
+  useEffect(() => {
+    getSysChecks()
+  }, [])
+
+  useEffect(() => {
+    if (!formik.values.stockCountId) {
+      setSiteStore([])
+      setControllerStore([])
+      setFilteredItems([])
+      setEditMode(false)
+    }
+  }, [formik.values.stockCountId])
+
+  const fillSiteStore = stockCountId => {
+    setSiteStore([])
+    setControllerStore([])
+    getRequest({
+      extension: SCRepository.Sites.qry,
+      parameters: `_stockCountId=${stockCountId}`
+    }).then(res => {
+      setSiteStore(res.list.filter(site => site.isChecked == true))
+    })
+  }
+
+  const fillControllerStore = (stockCountId, siteId) => {
+    getRequest({
+      extension: SCRepository.PHY.qry,
+      parameters: `_stockCountId=${stockCountId}&_siteId=${siteId}`
+    }).then(res => {
+      setControllerStore(res.list)
+    })
+  }
+
+  const checkPhyStatus = async controllerId => {
+    const resp = await getRequest({
+      extension: SCRepository.PHY.get,
+      parameters: `_stockCountId=${formik.values.stockCountId}&_siteId=${formik.values.siteId}&_controllerId=${controllerId}`
+    })
+
+    if (resp) {
```

Comment:
no need for this if condition, when updating value add nullable checks  `resp?.record?.status`